js/src/components/lineage/LineageView.tsx

```diff
@@ -735,15 +735,17 @@ export function PrivateLineageView(
       setFocusedNodeId(undefined);
       multiNodeAction.reset();
     } else {
-      setSelectedNodeIds((prev) => {
-        const newSelectedNodes = new Set(prev);
-        if (prev.has(nodeId)) {
-          newSelectedNodes.delete(nodeId);
-        } else {
-          newSelectedNodes.add(nodeId);
-        }
-        return newSelectedNodes;
-      });
+      const newSelectedNodeIds = new Set(selectedNodeIds);
+      if (selectedNodeIds.has(nodeId)) {
+        newSelectedNodeIds.delete(nodeId);
+      } else {
+        newSelectedNodeIds.add(nodeId);
+      }
+
+      setSelectedNodeIds(newSelectedNodeIds);
+      if (newSelectedNodeIds.size === 0) {
+        setSelectMode("single");
```

Comment:
Does `single` mode work like `init` mode now?

Reply:
I think it should also change. I will create another PR to address this problem.